server/service/src/requisition/request_requisition/delete.rs

```diff
@@ -94,13 +94,13 @@ fn validate(
         return Err(OutError::NotThisStoreRequisition);
     }
 
-    if requisition_row.status != RequisitionStatus::Draft {
-        return Err(OutError::CannotEditRequisition);
-    }
-
     if requisition_row.r#type != RequisitionType::Request {
         return Err(OutError::NotARequestRequisition);
     }
+
+    if requisition_row.status != RequisitionStatus::Draft {
+        return Err(OutError::CannotEditRequisition);
+    }
```

Comment:
I've rearranged error handling for mutating requisitions throughout. 

To me this makes more sense returning the wrong type of requisition error before returning a status error.